src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -330,15 +338,48 @@ internal PSVersionHashTable(IEqualityComparer equalityComparer) : base(equalityC
 
         /// <summary>
         /// Returns ordered collection with Keys of 'PSVersionHashTable'
+        /// We want see special order:
+        ///     1. PSVersionName
+        ///     2. PSEditionName
+        ///     3. Remaining properties in alphabetical order
         /// </summary>
         public override ICollection Keys
         {
             get
             {
-                Array arr = new string[base.Keys.Count];
-                base.Keys.CopyTo(arr, 0);
-                Array.Sort(arr, StringComparer.OrdinalIgnoreCase);
-                return arr;
+                ArrayList keyList = new ArrayList(base.Keys);
+                keyList.Sort(keysComparer);
+                return keyList;
+            }
+        }
+
+        private static PSVersionTableComparer keysComparer = new PSVersionTableComparer();
```

Comment:
Please make it `readonly`.
It's recommended to put field declarations before constructors, and for a private static field, it should be named with the prfix `s_` -- `s_keysComparer`. See `s_psVersionTable` in this file for an example.